### PR TITLE
test updates, clean ups

### PR DESCRIPTION
**Pull request by RagePit**
"forge test -f https://api.harmony.one/" or relevant RPC to run tests

### CODE REVIEW

File: src/LockedJewelOffer.sol [hunk 1]
```diff
@@ -17,11 +16,11 @@ interface IOwnable {
...
-    address public factory;
-    address public seller;
-    address public tokenWanted;
-    uint256 public amountWanted;
-    uint256 public fee; // in bps
+    address immutable public factory;
+    address immutable public seller;
+    address immutable public tokenWanted;
+    uint256 immutable public amountWanted;
+    uint256 immutable public fee; // in bps
```
Comment (RagePit) [inline src/LockedJewelOffer.sol:23]: Made these immutable to reinforce that the Offer cannot change

File: src/OfferFactory.sol [hunk 2]
```diff
@@ -26,20 +23,21 @@ contract OfferFactory is Ownable {
     }
 
     function getActiveOffersByOwner() public view returns (LockedJewelOffer[] memory, LockedJewelOffer[] memory) {
-        LockedJewelOffer[] memory myBids = new LockedJewelOffer[](offers.length);
-        LockedJewelOffer[] memory otherBids = new LockedJewelOffer[](offers.length);
+        uint256 offersLength = offers.length;
+
```
Comment (RagePit) [inline src/OfferFactory.sol:27]: Would like to change this to take address as param rather than use msg.sender

Comment (calebcheng00) [reply src/OfferFactory.sol:27]: just to be a bit more flexible in querying offers? not a big deal on the frontend.

File: src/LockedJewelOffer.sol (hunk 1 above)
Comment (calebcheng00) [reply src/LockedJewelOffer.sol:23]: sgtm

File: src/test/LockedJewelMarket.t.sol [hunk 3]
```diff
@@ -1,37 +1,119 @@
...
 
     function setUp() public {
-        factory = new OfferFactory(); // 250 bps, 2.5%
+        factoryDeployer = new FactoryDeployer();
+        factory = factoryDeployer.factory();
+
+        offerUser = new OfferUser();
+
+        //Give us 100k locked jewel to work with
+        VM.store(address(JEWEL), keccak256(abi.encode(address(this), 15)), bytes32(uint(100000 * 1e18)));
```
Comment (calebcheng00) [inline src/test/LockedJewelMarket.t.sol:42]: bless 🙏

**Review by calebcheng00 — COMMENTED** [review]
appreciate you - this is so awesome. 
will relay to @tdudz and get his comments as well. but lgtm

**Comment by tdudz** [comment]
@RagePit thanks for this! much appreciated. i had some cleanup of my own that i already pushed and it conflicted a bit with this. do you mind just pushing the test cases for now? ty 🙏

File: src/OfferFactory.sol (hunk 2 above)
Comment (tdudz) [reply src/OfferFactory.sol:27]: seems unnecessary to me, `offersLength` not much different from `offers.length` unless there's some gas optimization i'm not aware of (but even then, harmony is basically free)

**Comment by RagePit** [comment]
> @RagePit thanks for this! much appreciated. i had some cleanup of my own that i already pushed and it conflicted a bit with this. do you mind just pushing the test cases for now? ty 🙏

That's totally fine yeah

File: src/test/util/Vm.sol [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    // Calldata can either be strict or a partial match, e.g. if you only
+    // pass a Solidity selector to the expected calldata, then the entire Solidity
+    // function will be mocked.
+    function mockCall(address,bytes calldata,bytes calldata) external;
+    // Clears all mocked calls
+    function clearMockedCalls() external;
+    // Expect a call to an address with the specified calldata.
+    // Calldata can either be strict or a partial match
+    function expectCall(address,bytes calldata) external;
+}
```
Comment (tdudz) [inline src/test/util/Vm.sol:40]: add newline plz sir

**Pull request closed without merging**